From fc0c3636e5403b83a48621ed15dfbf6ec513c4b7 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 3 Feb 2017 11:16:21 +0800
Subject: [PATCH 1493/5242] MLK-13851-1 usb: chipidea: otg: delete the
 workaround code

commit  e4afee3fc17118a60ecc8d2d5f06c60ee8f5f5b5 from
https://source.codeaurora.org/external/imx/linux-imx.git

After commit 49670184289e ("usb: chipidea: otg: change workqueue
ci_otg as freezable"), we have fixed the bug that ID removed
wakeup (ID: 0->1) will lock up system resume, we delete the
workaround code in this commit.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/usb/chipidea/otg.c b/drivers/usb/chipidea/otg.c
index 8a9f120..b987472 100644
--- a/drivers/usb/chipidea/otg.c
+++ b/drivers/usb/chipidea/otg.c
@@ -3,6 +3,7 @@
  * otg.c - ChipIdea USB IP core OTG driver
  *
  * Copyright (C) 2013-2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP
  *
  * Author: Peter Chen
  */
@@ -212,12 +213,6 @@ void ci_handle_id_switch(struct ci_hdrc *ci)
 		dev_dbg(ci->dev, "switching from %s to %s\n",
 			ci_role(ci)->name, ci->roles[role]->name);
 
-		while (ci_hdrc_host_has_device(ci)) {
-			enable_irq(ci->irq);
-			usleep_range(10000, 15000);
-			disable_irq_nosync(ci->irq);
-		}
-
 		ci_role_stop(ci);
 
 		if (role == CI_ROLE_GADGET &&
-- 
1.7.9.5

